chia/rpc/wallet_rpc_api.py

```diff
@@ -1387,11 +1387,14 @@ async def dl_update_root(self, request) -> Dict:
 
         for _, wallet in self.service.wallet_state_manager.wallets.items():
             if WalletType(wallet.type()) == WalletType.DATA_LAYER:
-                record = await wallet.create_update_state_spend(
-                    bytes32.from_hexstr(request["launcher_id"]), bytes32.from_hexstr(request["new_root"])
+                records = await wallet.create_update_state_spend(
+                    bytes32.from_hexstr(request["launcher_id"]),
+                    bytes32.from_hexstr(request["new_root"]),
+                    fee=uint64(request.get("fee", 0)),
                 )
-                await self.service.wallet_state_manager.add_pending_transaction(record)
-                return {"tx_record": record.to_json_dict_convenience(self.service.config)}
+                for record in records:
+                    await self.service.wallet_state_manager.add_pending_transaction(record)
+                return {"tx_record": records[0].to_json_dict_convenience(self.service.config)}
```

Comment:
So we only return the "dl_tx" or the first item in this list but we call add_pending_transaction for all the records.

Reply:
Yeah I didn't want to break the RPCs that were currently in use.  I don't think which tx record you get is that important, but we can fix it later if it is.